_docker/drupal/drupal-filesystem/web/modules/custom/rhd_assemblies/src/Plugin/AssemblyBuild/DynamicContentFeedBuild.php

```diff
@@ -0,0 +1,143 @@
+
+    if (count($items)) {
+      $build['posts'] = [
+        '#theme' => 'item_list',
+        '#list_type' => 'ul',
+        '#items' => [],
+        '#attributes' => ['class' => 'content-tile-list'],
+      ];
+      foreach ($items as $item) {
+        if (isset($item['post'])) {
+          $build['posts']['#items'][] = [
+            '#theme' => 'wordpress_post_tile',
+            '#post' => $item['post']->content,
+            '#media' => $item['post']->media,
+            '#categories' => $item['post']->categories
+          ];
+        }
+        else if (isset($item['node'])) {
+          $view_builder = \Drupal::entityTypeManager()->getViewBuilder('node');
+          $storage = \Drupal::entityTypeManager()->getStorage('node');
+          $build['posts']['#items'][] = $view_builder->view($item['node'], 'tile');
+        }
+      }
+    }
+
+  }
+
+  protected function orderItems($posts, $nodes) {
+    $items = [];
+
+    // Combine into a single list, ordered by date. Use double array in case
+    // items share a creation time.
+    foreach ($posts as $post) {
+      $date = strtotime($post->content->date);
+      $items[$date][] = [
+        'post' => $post
+      ];
+    }
+    foreach ($nodes as $node) {
+      $date = $node->getCreatedTime();
+      $items[$date][] = [
+        'node' => $node
+      ];
+    }
+
+    // Flatten array
+    $flat_items = [];
+    foreach($items as $date) {
+      foreach ($date as $item) {
+        $flat_items[] = $item;
+      }
+    }
+
+    return $flat_items;
+  }
+
+  protected function getWordpressPosts(EntityInterface $entity, $count) {
+    $category_filters = $entity->get('field_category_filter')->getValue();
+    if (empty($category_filters)) {
+      return [];
+    }
+
+    $categories = [];
+    if (!empty($category_filters)) {
+      foreach ($category_filters as $category_filter) {
+        $categories[] = $category_filter['value'];
+      }
+    }
+
+    if (count($categories)) {
+      $posts = \Drupal::service('rhd_assemblies.wordpress_api')->getContentByCategory($categories, $count);
+      return $posts;
+    }
+
+    return [];
+  }
+
+  protected function getDrupalNodes(EntityInterface $entity, $count) {
+    $term_filters = $entity->get('field_drupal_term_filter')->getValue();
+    if (empty($term_filters)) {
+      return [];
+    }
+
+    $terms = [];
+    if (!empty($term_filters)) {
+      foreach ($term_filters as $term_filter) {
+        $terms[] = $term_filter['target_id'];
+      }
+    }
+
+    $valid_node_types = [
+      'video_resource'
+    ];
+
+    $query = \Drupal::database()->select('node__field_topics', 't');
+    $query->fields('t', ['entity_id']);
+    $query->condition('t.field_topics_target_id', $terms, 'in');
+    $query->leftJoin('node', 'n', 'n.nid = t.entity_id');
+    $query->condition('n.type', $valid_node_types, 'in');
+    $query->range(0, $count);
+    $query->orderBy('t.entity_id', 'desc');
+    $results = $query->execute();
```

Comment:
What are we doing with this select, and is it possible to do this with an entity query?